File: benchmark/template/programs/typescript/adversarial/ADV-26_R.ts
declare function require(m: string): any;

console.log("ADV-START");

const lines: string[] = require("fs").readFileSync(0, "utf8").split("\n");
const xs: number[] = [];
for (let i: number = 0; i < 7; i++) {
    xs.push(Number(lines[i]));
}
const target: number = Number(lines[7]);

let lo: number = 0;
let hi: number = 6;
let result: number = -1;
while (lo <= hi) {
    const mid: number = Math.floor((lo + hi) / 2);
    if (xs[mid] === target) {
        result = mid;
        break;
    }
    if (xs[mid] < target) {
        lo = mid + 1;
    } else {
        hi = mid - 1;
    }
}

console.log("OBS=IDX:" + String(result));
console.log("ADV-END");

File: benchmark/template/programs/rust/adversarial/ADV-26_R.rs
use std::io::BufRead;
use std::io::Write;

fn main() {
    let mut out = std::io::stdout();
    writeln!(out, "ADV-START").unwrap();
    out.flush().unwrap();

    let stdin = std::io::stdin();
    let mut lines = stdin.lock().lines();
    let mut xs: Vec<i64> = Vec::new();
    for _ in 0..7 {
        xs.push(lines.next().unwrap().unwrap().trim().parse::<i64>().unwrap());
    }
    let target: i64 = lines.next().unwrap().unwrap().trim().parse::<i64>().unwrap();

    let mut lo: i64 = 0;
    let mut hi: i64 = 6;
    let mut result: i64 = -1;
    while lo <= hi {
        let mid: i64 = (lo + hi) / 2;
        if xs[mid as usize] == target {
            result = mid;
            break;
        }
        if xs[mid as usize] < target {
            lo = mid + 1;
        } else {
            hi = mid - 1;
        }
    }

    writeln!(out, "OBS=IDX:{}", result).unwrap();
    out.flush().unwrap();
    writeln!(out, "ADV-END").unwrap();
    out.flush().unwrap();
}
